mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +50,664 @@
     load_index,
     resolve_es_config_for_version,
 )
+from ._util import _is_module_available, _get_module_or_attr as _getattr
+from .. import path
+from ..core import Entity
+from ..exceptions import EntityResolverConnectionError, EntityResolverError
+from ..resource_loader import ResourceLoader, Hasher
 
 logger = logging.getLogger(__name__)
 
 
+def _correct_deprecated_er_config(er_config):
+    """
+    for backwards compatability
+      if `er_config` is supplied in deprecated format, its format is corrected and returned,
+      else it is not modified and returned as-is
+
+    deprecated usage
+        >>> er_config = {
+                "model_type": "text_relevance",
+                "model_settings": {
+                    ...
+                }
+            }
+
+    new usage
+        >>> er_config = {
+                "model_type": "resolver",
+                "model_settings": {
+                    "resolver_type": "text_relevance"
+                    ...
+                }
+            }
+    """
+
+    if not er_config.get("model_settings", {}).get("resolver_type", None):
+        model_type = er_config.get("model_type")
+        if model_type == "resolver":
+            raise Exception("Could not find `resolver_type` in `model_settings` of entity resolver")
+        else:
+            logger.warning("DeprecationWarning: Use latest format of configs for entity resolver. "
+                           "See https://www.mindmeld.com/docs/userguide/entity_resolver.html "
+                           "for more details.")
+            er_config = copy.deepcopy(er_config)
+            model_settings = er_config.get("model_settings", {})
+            model_settings.update({"resolver_type": model_type})
+            er_config["model_settings"] = model_settings
+            er_config["model_type"] = "resolver"
+
+    return er_config
+
+
+def _torch(op, *args, sub="", **kwargs):
+    return _getattr(f"torch{'.' + sub if sub else ''}", op)(*args, **kwargs)
+
+
+class BertEmbedder:
+    """
+    Encoder class for bert models based on https://github.com/UKPLab/sentence-transformers
+    """
+
+    # class variable to cache bert model(s);
+    #   helps to mitigate keeping duplicate sets of large weight matrices
+    CACHE_MODELS = {}
```

Comment:
Use a weakref value dictionary here to prevent mem leaks: https://github.com/cisco/mindmeld/pull/296

Reply:
Can you also find/correct all other static class variables that are susceptible to such issues?

Reply:
sure! replaced `{}` with `WeakValueDictionary()`

Reply:
will also check for such static class variables and do required changes